src/pal/src/thread/thread.cpp

```diff
@@ -2505,89 +2505,109 @@ ThreadInitializationRoutine(
     return NO_ERROR;
 }
 
-// Get base address of this thread's stack
-// Can be called only for the current thread.
+// Get base address of the current thread's stack
 void *
 CPalThread::GetStackBase()
 {
-    _ASSERT_MSG(this == InternalGetCurrentThread(), "CPalThread::GetStackBase called from foreign thread");
```

Comment:
I have not removed it - it is still in the GetCachedStackBase, which replaced the GetStackBase. The new GetStackBase is a static method that reads the values for the current thread from pthreads.